treeherder/seta/job_priorities.py

```diff
@@ -0,0 +1,141 @@
+            jp.testtype = jp.testtype.replace('e10s-devtools-chrome', 'devtools-chrome-e10s')
+            jp.testtype = jp.testtype.replace('gl-', 'webgl-')
+
+        key = jp.unique_identifier()
+        if key in ref_data_names:
+            # e.g. desktop-test-linux64-pgo/opt-reftest-13 or builder name
+            jobs.append(ref_data_names[key])
+        else:
+            logger.warning('We did not find job priority ({}) in the list of accepted jobs'.format(jp))
+
+    return jobs
+
+
+def _gecko_decision_task_request(project):
+    # In the case of a request from the Gecko decision task we don't care which priority is
+    # requested. We return either all jobs every 5th push or 90 minutes for that project or
+    # all TaskCluster high value jobs.
+    task_request = _update_task_request(project)
+    last_reset = task_request.seconds_since_last_reset()
+    job_priorities = []
+    for jp in _query_job_priorities(priority=0, excluded_build_system_type='buildbot'):
+        # We only add a job if it hasn't reached its timeout.
+        # If the timeout is zero it means that the job needs to always run
+        if last_reset < jp.timeout or jp.timeout == 0:
+            # Due to the priority of all high value jobs is 1, and we
+            # We need to return all jobs for every 5th push.
+            if task_request.counter % jp.priority != 0:
+                job_priorities.append(jp)
+
+    return _process(build_system='taskcluster', job_priorities=job_priorities)
+
+
+def _update_task_request(project):
+    '''Update a repository's counter and/or reset timer.'''
+    now = datetime.datetime.now()
+
+    try:
+        task_request = TaskRequest.objects.get(repository__name=project)
+        # Increase the counter of how many times the gecko decision task
+        # has contacted us
+        task_request.counter += 1
+        if task_request.has_expired():
+            task_request.last_reset = now
+        task_request.save()
+    except TaskRequest.DoesNotExist:
+        task_request = TaskRequest.objects.create(
+            repository=Repository.objects.get(name=project),
+            counter=1,
+            last_reset=now,
+            reset_delta=RESET_DELTA)
+
+    return task_request
+
+
+def _query_job_priorities(priority, excluded_build_system_type):
+    job_priorities = JobPriority.objects.all()
+    if priority != 0:
+        job_priorities = job_priorities.filter(priority=priority)
+    return job_priorities.exclude(buildsystem=excluded_build_system_type)
+
+
+def _validate_request(build_system_type, project):
+    if build_system_type not in ('buildbot', 'taskcluster', '*'):
+        raise SetaError('Valid build_system_type values are buildbot or taskcluster.')
+
+
+def seta_job_scheduling(project, build_system_type, priority=5, user_agent=None):
+    _validate_request(build_system_type, project)
+
+    if user_agent == GECKO_DECISION_TASK_USER_AGENT:
```

Comment:
It is an increment.
The test can help understand:
https://github.com/mozilla/treeherder/pull/1957/files#diff-b13ce1b054e5882fab875f0054aa20c7R34

The Gecko decision task does *not* set the value (thus, a POST would not make sense) *but* the call in itself increases the counter.

I will change now to a url parameter for now. I understand the concern of abusing it. I will follow up this coming week before we use it in production.

Could we follow up on a new bug to do the following?

- Give a Hawk key to the Gecko decision task
   - I will need technical details on how this is done

Can an endpoint behave differently depending if a GET call has a Hawk API key? or should we use a different endpoint for it?